ibis/pandas/execution/generic.py

```diff
@@ -462,6 +471,20 @@ def execute_count_distinct_series_mask(op, data, mask, context=None, **kwargs):
     return context.agg(data[mask] if mask is not None else data, 'nunique')
 
 
+@execute_node.register(ops.Arbitrary, pd.Series, (pd.Series, type(None)))
+def execute_arbitraryt_series_mask(op, data, mask, context=None, **kwargs):
+    if op.how == 'first':
+        index = 0
+    elif op.how == 'last':
+        index = -1
+    else:
+        raise com.OperationNotDefinedError(
+            'Arbitrary {} is not supported'.format(op.how))
+
+    data = data[mask] if mask is not None else data
+    return data.iloc[index]
```

Comment:
I think you need to use context here, so that `arbitrary` can be used as a window function. There should be a bunch of examples of how to use it in this file.

Reply:
Series `first` and `last` methods are not working here https://pandas.pydata.org/pandas-docs/stable/generated/pandas.Series.first.html

Am I using the wrong signature? I thought it was for `table.column.reducer()` scenario.

```python
@execute_node.register(ops.Arbitrary, pd.Series, (pd.Series, type(None)))
```

Reply:
The case I'm thinking of is something like:

```python
t.a - t.a.arbitrary()
```

where the result of `t.a.arbitrary()` needs to be projected over the length of `t.a`

Reply:
It's fine to not use `first` and `last`

Reply:
@cpcloud could You give me a full expression I should test against it?